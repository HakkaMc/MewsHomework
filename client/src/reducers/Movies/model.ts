export interface Movie {
	id: number;
	original_language: string;
	original_title: string;
	poster_path: string;
	title: string;
	vote_average: number;
	vote_count: number;
}

export interface Movies {
	movies?: Array<Movie>;
	page: number;
	selectedMovie?: Movie;
	totalResults: number;
	totalPages: number;
	searchMoviesAjaxId: string;
	getMovieDetailAjaxId: string;
}

const initialState: Movies = {
	movies: undefined,
	page: 0,
	selectedMovie: undefined,
	totalResults: 0,
	totalPages: 0,
	searchMoviesAjaxId: "",
	getMovieDetailAjaxId: ""
};

export default initialState;
