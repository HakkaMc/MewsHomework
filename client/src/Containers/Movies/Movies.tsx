import React from "react";
import { Grid } from "@material-ui/core";

import ResultList from "./Components/ResultList/ResultList";
import Detail from "./Components/Detail/Detail";
import Pagination from "./Components/Pagination/Pagination";

export const Movies: React.FC = () => {
	return (
		<>
			<Pagination />
			<Grid container spacing={3} style={{ paddingTop: "10px" }}>
				<Grid item xs={12} sm={6}>
					<ResultList />
				</Grid>
				<Grid item xs={12} sm={6}>
					<Detail />
				</Grid>
			</Grid>
			<Pagination />
		</>
	);
};

export default Movies;
