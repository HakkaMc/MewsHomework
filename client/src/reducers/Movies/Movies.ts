import produce, { Draft } from "immer";
import initialState, { Movies, Movie } from "./model";
import { ReduxAction, SetMovies } from "../../custom";
import { action, processActions } from "../../utils/redux";

export const moviesActions = {
	clearAll: () => action(),
	clearSelectedMovie: () => action(),
	getMovieDetail: (movieId: number) => action(movieId),
	searchMovies: (movieName?: string, pageNumber?: number) =>
		action({ movieName, pageNumber }),
	setMovies: (movies: SetMovies) => action(movies),
	setSelectedMovie: (movie: Movie) => action(movie),
	setSearchMoviesAjaxId: (ajaxId: string) => action(ajaxId),
	setGetMovieDetailAjaxId: (ajaxId: string) => action(ajaxId)
};

processActions(moviesActions, "MOVIES");

export default produce<any, any, Movies>(
	(state: Draft<Movies>, action: ReduxAction): Draft<Movies> => {
		switch (action.type) {
			case moviesActions.clearAll.toString(): {
				state = { ...initialState };
				break;
			}

			case moviesActions.clearSelectedMovie.toString(): {
				state.selectedMovie = undefined;
				break;
			}

			case moviesActions.setMovies.toString(): {
				const payload: SetMovies = action.payload;
				state.movies = payload.movies;
				state.page = payload.page;
				state.totalPages = payload.totalPages;
				state.totalResults = payload.totalResults;
				break;
			}

			case moviesActions.setSelectedMovie.toString(): {
				const payload: Movie = action.payload;
				state.selectedMovie = payload;
				break;
			}

			case moviesActions.setSearchMoviesAjaxId.toString(): {
				state.searchMoviesAjaxId = action.payload;
				break;
			}

			case moviesActions.setGetMovieDetailAjaxId.toString(): {
				state.getMovieDetailAjaxId = action.payload;
				break;
			}

			default:
		}

		return state;
	},
	initialState
);
