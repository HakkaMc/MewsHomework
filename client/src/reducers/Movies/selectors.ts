import { ReduxState } from "../index";
import get from "lodash/get";

export const selectedMovieName = (state: ReduxState): string =>
	get(state, "form.search.values.movieName") || "";

export const selectSearchMoviesAjaxId = (state: ReduxState): string =>
	state.movies.searchMoviesAjaxId;
export const selectGetMovieDetailAjaxId = (state: ReduxState): string =>
	state.movies.getMovieDetailAjaxId;
