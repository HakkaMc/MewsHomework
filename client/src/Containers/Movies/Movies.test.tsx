import React from "react";
import { shallow } from "enzyme";

import { Movies } from "./Movies";

describe("<Movies/>", () => {
	it("Match snapshot", () => {
		const wrapper = shallow(<Movies />);
		expect(wrapper).toMatchSnapshot();
	});
});
