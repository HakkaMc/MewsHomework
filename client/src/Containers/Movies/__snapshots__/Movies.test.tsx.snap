// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies/> Match snapshot 1`] = `
<Fragment>
  <Connect(Pagination) />
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
    style={
      Object {
        "paddingTop": "10px",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <Connect(ResultList) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <Connect(Detail) />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <Connect(Pagination) />
</Fragment>
`;
